File: deal/src/main/java/ru/artemev/deal/dto/EmploymentDTO.java
package ru.artemev.deal.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Builder;
import lombok.Data;
import ru.artemev.deal.model.enums.EmploymentStatus;
import ru.artemev.deal.model.enums.Position;

import java.math.BigDecimal;

@Data
@Builder
@Schema(description = "Трудоустройство")
@JsonDeserialize(builder = EmploymentDTO.EmploymentDTOBuilder.class)
public class EmploymentDTO {
  @Schema(description = "Оклад", example = "15000")
  @JsonProperty("salary")
  private final BigDecimal salary;

  @Schema(description = "Должность", example = "MID_MANAGER")
  @JsonProperty("position")
  private final Position position;

  @Schema(description = "Общий трудовой стаж", example = "12")
  @JsonProperty("workExperienceTotal")
  private final Integer workExperienceTotal;

  @Schema(description = "Текущий трудовой стаж", example = "3")
  @JsonProperty("workExperienceCurrent")
  private final Integer workExperienceCurrent;

  @Schema(description = "Рабочий статус", example = "SELF_EMPLOYED")
  @JsonProperty("employmentStatus")
  private EmploymentStatus employmentStatus;

  @Schema(description = "ИНН", example = "123456")
  @JsonProperty("employerINN")
  private String employerINN;
}

File: gateway/src/main/kotlin/ru/artemev/gateway/dto/EmploymentDTO.kt
package ru.artemev.gateway.dto

import io.swagger.v3.oas.annotations.media.Schema
import ru.artemev.gateway.model.enums.EmploymentStatus
import ru.artemev.gateway.model.enums.Position
import java.math.BigDecimal

@Schema(description = "Трудоустройство")
data class EmploymentDTO(
    @Schema(description = "Оклад", example = "15000")
    val salary: BigDecimal? = null,

    @Schema(description = "Должность", example = "MID_MANAGER")
    val position: Position? = null,

    @Schema(description = "Общий трудовой стаж", example = "12")
    val workExperienceTotal: Int? = null,

    @Schema(description = "Текущий трудовой стаж", example = "3")
    val workExperienceCurrent: Int? = null,

    @Schema(description = "Рабочий статус", example = "SELF_EMPLOYED")
    val employmentStatus: EmploymentStatus? = null,

    @Schema(description = "ИНН", example = "123456")
    val employerINN: String? = null
)
